### PR TITLE
119 firebase emulation

### LINKED ISSUE #119: Misc.12 Set up environment files for using Firebase local emulator
We want to be able to run firebase locally during development

- [x] Update app.config and environment files to support easy local firebase emulation.
- [x] Create a tutorial in the discussions

### PR DESCRIPTION
**Pull request by ashkjones**
Fixes #119 
- I put the steps to running the emulator in discussion #120 
- This will help people test their code using the Firestore service without having to ping the live server
- I recommend still using the json-server for development if you are on a less powerful laptop because the Firebase emulator may be slow

### CODE REVIEW

File: src/app/app.config.ts [hunk 1]
```diff
@@ -6,27 +6,49 @@ import { HttpClientModule } from '@angular/common/http';
 import { StudySetDevService } from './services/study-set-dev.service';
 import { initializeApp, provideFirebaseApp } from '@angular/fire/app';
 import { getAuth, provideAuth } from '@angular/fire/auth';
-import { getFirestore, provideFirestore } from '@angular/fire/firestore';
-import { getStorage, provideStorage } from '@angular/fire/storage';
-import { environment } from '../environments/environment.development';
+import { connectFirestoreEmulator, getFirestore, provideFirestore } from '@angular/fire/firestore';
+import { connectStorageEmulator, getStorage, provideStorage } from '@angular/fire/storage';
+import { environment } from '../environments/environment';
```
Comment (B-Yamawaki) [inline src/app/app.config.ts:11]: why was this changed from environment.development to environment? The variable, useFirebase could be changed in environment.ts, but I think that would ignore the point of the development environment. It might've just been a change that was overlooked.

Comment (ashkjones) [reply src/app/app.config.ts:11]: The `environment` file gets replaced when the code is run in the development environment (i.e., ng serve) by the configs in `angular.json`
https://github.com/PonderTeam/ponder-app/blob/7dcd998da98c32b96830bafb87b77c0e53f80ce9/angular.json#L58-L66

If you run the code and look in the browser console, you should see "Angular is running in development mode."

## with `useFirebase: true`
image

## with `useFirebase: false`
image

Comment (ashkjones) [reply src/app/app.config.ts:11]: You can also click on the sets and confirm that the id's in the url's don't match the the id's in the cloud Firestore. Therefore, we can safely assume that one of the local data sources is being used instead.

Comment (B-Yamawaki) [reply src/app/app.config.ts:11]: I see. I didn't know that. Thank you for clarifying.

**Review by B-Yamawaki — APPROVED** [review]
Looks good. Probably makes it easier to make a release
